packages/editor/templates/Coin/contracts/Coin.sol

```diff
@@ -1,35 +1,56 @@
+// Specifies that the source code is for a version
+// of Solidity greater than 0.5.10
 pragma solidity >=0.5.0 <0.7.0;
 
+// A contract is a collection of functions and data (its state)
+// that resides at a specific address on the Ethereum blockchain.
 contract Coin {
-    // The keyword "public" makes variables
-    // accessible from other contracts
+    // The keyword "public" makes variables accessible from outside a contract
+    // and creates a function that other contracts or SDKs can call to access the value
+
+    // An address stores addresses of contracts, or a hash of the public half
+    // of a keypair belonging to external accounts
     address public minter;
+
+    // A mapping lets you create complex custom data types.
+    // This mapping assigns an unsigned integer to an address
+    // and is also a public variable.
     mapping (address => uint) public balances;
 
-    // Events allow clients to react to specific
-    // contract changes you declare
+    // Events allow Ethereum clients to react to specific
+    // contract changes you declare.
+    // This defines the event and it is sent later
     event Sent(address from, address to, uint amount);
 
-    // Constructor code is only run when the contract
-    // is created
+    // A special function only run during the creation of the contract
     constructor() public {
+        // Uses the special msg global variable to store the
+        // address of the contract creator
         minter = msg.sender;
     }
 
     // Sends an amount of newly created coins to an address
-    // Can only be called by the contract creator
     function mint(address receiver, uint amount) public {
+        // require statements define conditions that must pass
+        // before state is changed
+
+        // Can only be called by the contract creator
```

Comment:
Perhaps worth explaining a revert if a `require` fails? Up to you 😄

Reply:
You mean in terms that the state reverts?

Reply:
Exactly, that when a `require` fails (evaluates to false), it triggers an exception that reverts all modifications done to the state in the current call, including all sub-calls & shows an error to the caller.